### ISSUE TITLE
LXR | Unable to lex characters: '10_000,' - postgres dialect fails on underscores in numbers

### ISSUE THREAD

**Issue by bjmc** [sqlfluff/sqlfluff#7369]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

As of PostgreSQL 16, [underscores are supported in numeric constants](https://www.depesz.com/2023/02/06/waiting-for-postgresql-16-allow-underscores-in-integer-and-numeric-constants/) to help make big numbers easier to read, but SQLFluff chokes on numbers with underscores in them.



### Expected Behaviour

Expected valid Postgres SQL to be lexed/parsed by SQLFluff

### Observed Behaviour

SQLFluff fails with output like:

```
 FAIL                                                                                                                                                                                                   
L:  32 | P:   1 |  PRS | Line 32, Position 1: Found unparsable section: 'CREATE                                                                                                                                                                                                    
                       | TABLE status.valid_statuses\n(\n   ...'
L:  42 | P:   3 |  LXR | Unable to lex characters: '10_000,'
L:  43 | P:   3 |  LXR | Unable to lex characters: '7_000,'
L:  44 | P:   3 |  LXR | Unable to lex characters: '5_000,'
L:  45 | P:   3 |  LXR | Unable to lex characters: '3_000,'
L:  46 | P:   3 |  LXR | Unable to lex characters: '1_000,'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly? 
```

### How to reproduce

```sql
CREATE TABLE test
(
    val smallint PRIMARY KEY,
    label text NOT NULL,
);

INSERT INTO test (val, label)
VALUES
(10_000, "NEW"),
(1_000, "OLD");
```

### Dialect

postgresql

### Version

```
$ sqlfluff --version
sqlfluff, version 3.5.0
python --version
Python 3.10.12
```



### Configuration

```toml
[tool.sqlfluff.core]
dialect = "postgres"
max_line_length = 90

[tool.sqlfluff.indentation]
 # https://docs.sqlfluff.com/en/latest/configuration/layout.html#implicit-indents
implicit_indents = "allow"
indented_on_contents = false

[tool.sqlfluff.rules.references.keywords]
# Non-reserved https://www.postgresql.org/docs/current/sql-keywords-appendix.html
ignore_words = "data, version, name, path, status, cycle, comment, comments"

[tool.sqlfluff.rules.layout.long_lines]
ignore_comment_lines = true

[tool.sqlfluff.rules.capitalisation.functions]
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.types]
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.literals]
extended_capitalisation_policy = "lower"
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by paultiq** [comment]
Related to #7369

**Comment by bjmc** [comment]
@paultiq typo? This is #7369

**Comment by paultiq** [comment]
Thank you... I meant: #7375
